### PR TITLE
Improved locators from the different pages

### PR DESCRIPTION
**Pull request by danielgalvis98**
In the comments, I explain why I chose to change the locators or leave them as they are. Also, when modified, I say why I think that it is convenient for why to locate the elements, but I know that it is something that can be discussed.

### CODE REVIEW

File: src/page/address-step.page.ts
Comment (yesid-lopez) [inline src/page/address-step.page.ts]: maybe `form` and `span` is not needed

Comment (holgiosalos) [reply src/page/address-step.page.ts]: or maybe, there is an attribute that makes this button unique ;)

Comment (danielgalvis98) [reply src/page/address-step.page.ts]: > maybe `form` and `span` is not needed

Yeah, maybe on some locator locators I put some extra elements trying to avoid the problem that I mentioned on Slack, but I will be more careful and try to reduce this one and the others that may have more elements than needed.

Respecting the comments, I know that on a "real" code that shouldn't be there jeje, but Holmes said on Slack that we could put the justifications on GitHub or as code comments, so I made it that way just to ease things and with purposes of the exercise, but I know that it is not a good idea on a real project.

File: src/page/bank-payment.page.ts
Comment (holgiosalos) [inline src/page/bank-payment.page.ts]: you can play a little bit more, try to use a different locator, for example, the text related with the button seems to be unique "I confirm my order". Find a locator in where you can use that text.

File: src/page/order-summary.page.ts
Comment (holgiosalos) [inline src/page/order-summary.page.ts]: maybe the  `> strong` is not needed

File: src/page/sign-in-step.page.ts
Comment (holgiosalos) [inline src/page/sign-in-step.page.ts]: maybe the span is not needed since id is enough

File: src/page/product-list.page.ts
Comment (holgiosalos) [inline src/page/product-list.page.ts]: not sure about the data-id-product, what about the title attribute. There are two elements that share the data-id-produt attribute
image

File: src/page/shipping-step.page.ts
Comment (holgiosalos) [inline src/page/shipping-step.page.ts]: in my opinion, the original locator is fine, since it is the id.

File: src/page/shipping-step.page.ts
Comment (holgiosalos) [inline src/page/shipping-step.page.ts]: I think that just `button[name="processCarrier"]` or `[name="processCarrier"]` will be enough

File: src/page/address-step.page.ts
Comment (holgiosalos) [reply src/page/address-step.page.ts]: yes, my fault, but you can remove your comments on final commit after you get the approval for this PR

File: src/page/bank-payment.page.ts
Comment (danielgalvis98) [reply src/page/bank-payment.page.ts]: I was thinking that it is not a good idea to locate an element by his related text because it is likely to be changed at any moment for different reasons (for example, the page is being translated) so I avoided to use the content of elements as locators, but I can try it anyway.

File: src/page/order-summary.page.ts
Comment (danielgalvis98) [reply src/page/order-summary.page.ts]: Yeah, I didn't check that the p was just for that title and thought that the "Please send us a bank wire with" was part of it, I need to change my glasses... Will fix it.

File: src/page/product-list.page.ts
Comment (danielgalvis98) [reply src/page/product-list.page.ts]: To be honest I didn't see that there was another element with the same data-id-product attribute, but that is what I was trying to avoid setting that it has to make part of ul.product_list. But thinking more, maybe it is not a good idea as you say because they could also put another option inside the same list using again the data-id-product attribute.

But if I choose to use the title attribute (which is Add to cart), if some other product is added I think there will be a problem, don't you think?

Comment (holgiosalos) [reply src/page/product-list.page.ts]: That's right, I mean, use the ul.product_list and the title attribute together

File: src/page/bank-payment.page.ts
Comment (holgiosalos) [reply src/page/bank-payment.page.ts]: Yeah you're right, but It is a supposition, in real projects that kind of requirements are clear. I want to you explore more types of locators besides css selectors. CSS Selectors are not a golden hammer.

File: src/page/address-step.page.ts
Comment (danielgalvis98) [reply src/page/address-step.page.ts]: I will do it.

File: src/page/bank-payment.page.ts
Comment (danielgalvis98) [reply src/page/bank-payment.page.ts]: Fixed using element(by.buttonText...). Interesting...

File: src/page/sign-in-step.page.ts
Comment (danielgalvis98) [reply src/page/sign-in-step.page.ts]: It's true. I removed it.

File: src/page/product-list.page.ts
Comment (danielgalvis98) [reply src/page/product-list.page.ts]: I get it. Done.

File: src/page/shipping-step.page.ts
Comment (danielgalvis98) [reply src/page/shipping-step.page.ts]: It's true. fixed.

File: src/page/shipping-step.page.ts
Comment (danielgalvis98) [reply src/page/shipping-step.page.ts]: It's true. I thought maybe it was a good idea to leave at least te p.cart_navgitation since another button with the same name could appear at some other place (Like happens on the summary step page), but in the end I decided to leave just the button[name="processCarrier"]. What is the best practice in that case?

**Review by holgiosalos — APPROVED** [review]
Locators looks good to me, please remove all justification comments. Then you can squash and merge 😄

**Pull request merged**
